### PR TITLE
parse_result: provide badge svg

### PR DESCRIPTION
**Pull request by miri64**
This generates a little badge for every build that signifies the build results. This can be shown e.g. on the website or in the GitHub README to signify the status of the nightly builds to users.

### CODE REVIEW

**Comment by miri64** [comment]
See e.g. https://github.com/nextcloud/docker#what-is-nextcloud for an example

**Comment by miri64** [comment]
These are the two types of badges generated:

CI builds and tests passed

CI builds or tests failed

File: parse_result.py [hunk 1]
```diff
@@ -130,13 +154,16 @@ def static():
...
 
     static_tests = result_dict.pop("static_tests", {})
-    if nfailed:
-        print("--- result: BUILD FAILED!")
-        if html and ((nfailed > 1) or has_passed(static_tests)):
-            print("\n--- ", end="")
-            print(html_link("#error0", "JUMP TO FIRST ERROR OUTPUT"))
-    else:
-        print("--- result: BUILD SUCCESSFUL.")
+    with open("badge.svg", "w") as badge_svg:
```
Comment (kaspar030) [inline parse_result.py]: How about factoring out the badge writing?

```
def badge_write(filename, status="unknown"):
 with open(filename, "w") as badge_svg:
   badge_svg.write(badge.substitute(status=status))
```

... then call that within ```nfailed``` or else?
Feels weird to have the stdout printing within the "with".

**Review by kaspar030 — CHANGES_REQUESTED** [review]
Nice!

File: parse_result.py (hunk 1 above)
Comment (miri64) [reply parse_result.py]: Done.

**Comment by miri64** [comment]
Something still missing?

**Comment by kaspar030** [comment]
> Something still missing?

No, please squash! Mind you, this will not be live until me or @smlng find time to update murdock's scripts manually.

**Comment by miri64** [comment]
Squashed and acknowledged. Just let me know when you do, so we can update RIOT's README.

**Comment by miri64** [comment]
Do I need to wait for an official ACK in this repo or can I just hit merge?

File: parse_result.py [hunk 2]
```diff
@@ -103,6 +103,37 @@ def has_passed(job):
...
+<rect rx="3" x="24" width="53" height="20" class="${status}" />
+<rect x="24" width="4" height="20" class="${status}" />
+<g fill="#fff" text-anchor="middle" font-family="DejaVu Sans,Verdana,Geneva,sans-serif" font-size="11">
+  <text x="19" y="14">CI</text>
+  <text x="50" y="14">${status}</text>
+</g>
+</svg>
+""")
+    with open(filename, "w") as badge_svg:
+        badge.substitute(status=status)
```
Comment (kaspar030) [inline parse_result.py]: I think there's a "write()" missing

**Review by kaspar030 — CHANGES_REQUESTED** [review]

File: parse_result.py (hunk 2 above)
Comment (miri64) [reply parse_result.py]: Oopsie

File: parse_result.py [hunk 3]
```diff
@@ -103,6 +103,37 @@ def has_passed(job):
         return False
     return job["result"]["status"] in { 0, "0", "pass" }
 
+
+def create_badge(filename, status="failed"):
```
Comment (miri64) [inline parse_result.py:107]: The badge should be able to handle states other than `failed` and `passed`, but ok ;-).

**Comment by kaspar030** [comment]
@miri64 I just fixed the missing write, also reverted the default for "create_badge" to failed, as "unknown" is not a valid class in the badge's svg.

**Review by kaspar030 — APPROVED** [review]

**Comment by miri64** [comment]
> as "unknown" is not a valid class in the badge's svg.

A missing class just evaluates to the default style (which was intended) ;-)

**Comment by kaspar030** [comment]
> A missing class just evaluates to the default style (which was intended) ;-)

ok, too careful, sorry! :)

**Comment by kaspar030** [comment]
Alright, this is live. See e.g., here: https://ci.riot-os.org/RIOT-OS/RIOT/8788/eaea0048f800d72731b5bf192f26d840ef2d5443/badge.svg

**Comment by miri64** [comment]
Something is wrong with the `CI` part, though :-/

**Comment by miri64** [comment]
See #15.

**Pull request merged**